### PR TITLE
Unusable pixels in dl1ab

### PR DESCRIPTION
**Pull request by pawel21**
Fix Unusable pixels error in dl1ab #669.

According to @FrancaCassol and @moralejo suggestion, I took calibration_id from `f.root.dl1.event.telescope.monitoring.pedestal.col('calibration_id')` to got unusable_pixels.

This solve problem with production, but such data  as run 02927, sub-run 0164  have wrong interleaved pedestal (pedestal from calibration run instead from data run). Do you have any idea how to solve it?

### CODE REVIEW

File: lstchain/calib/camera/pixel_threshold_estimation.py [hunk 1]
```diff
@@ -62,17 +62,18 @@ def get_threshold_from_dl1_file(dl1_path, sigma_clean):
...
     # return pedestal interleaved threshold from data run for high gain
     return threshold_clean_pe[interleaved_events_id, HIGH_GAIN, :]
 
-def get_unusable_pixels(dl1_path, interleaved_events_id):
+def get_unusable_pixels(dl1_path):
     with tables.open_file(dl1_path) as f:
+        calibration_id = f.root.dl1.event.telescope.monitoring.pedestal.col('calibration_id')
         unusable_pixels = np.where(f.root[dl1_params_tel_mon_cal_key].col(
-                                   'unusable_pixels')[interleaved_events_id,
+                                   'unusable_pixels')[calibration_id[1],
```
Comment (maxnoe) [inline lstchain/calib/camera/pixel_threshold_estimation.py:76]: Why the hardcoded 1 here?

Comment (pawel21) [reply lstchain/calib/camera/pixel_threshold_estimation.py:76]: as I understood `1` in `calibration_id = f.root.dl1.event.telescope.monitoring.pedestal.col('calibration_id')` correspond to id for interleaved events.

Comment (maxnoe) [reply lstchain/calib/camera/pixel_threshold_estimation.py:76]: Then please define a constant `INTERLEAVED_CALIBRATION_ID`

Comment (pawel21) [reply lstchain/calib/camera/pixel_threshold_estimation.py]: Done

File: lstchain/calib/camera/pixel_threshold_estimation.py [hunk 2]
```diff
@@ -49,30 +50,31 @@ def get_threshold_from_dl1_file(dl1_path, sigma_clean):
...
     ped_mean_pe, ped_std_pe = get_bias_and_std(dl1_path)
 
     # If problem with interleaved pedestal std values occur, take pedestal
     # std values from calibration run.
     # Correct interleaved pedestal std array should have shape (2,2,1855)
     if ped_std_pe.shape[0] == 2:
-        interleaved_events_id = 1
+        interleaved_events_id = INTERLEAVED_CALIBRATION_ID
     else:
         interleaved_events_id = 0
```
Comment (moralejo) [inline lstchain/calib/camera/pixel_threshold_estimation.py:62]: For consistency, you should define   ORIGINAL_CALIBRATION_ID = 0   (or whatever other name)

Comment (moralejo) [inline lstchain/calib/camera/pixel_threshold_estimation.py:62]: This is contradictory, if it is 0 it is not interleaved

File: lstchain/calib/camera/pixel_threshold_estimation.py [hunk 3]
```diff
@@ -49,30 +50,31 @@ def get_threshold_from_dl1_file(dl1_path, sigma_clean):
...
     # return pedestal interleaved threshold from data run for high gain
     return threshold_clean_pe[interleaved_events_id, HIGH_GAIN, :]
 
-def get_unusable_pixels(dl1_path, interleaved_events_id):
+def get_unusable_pixels(dl1_path):
     with tables.open_file(dl1_path) as f:
+        calibration_id = f.root.dl1.event.telescope.monitoring.pedestal.col('calibration_id')
         unusable_pixels = np.where(f.root[dl1_params_tel_mon_cal_key].col(
-                                   'unusable_pixels')[interleaved_events_id,
+                                   'unusable_pixels')[calibration_id[INTERLEAVED_CALIBRATION_ID],
```
Comment (moralejo) [inline lstchain/calib/camera/pixel_threshold_estimation.py:77]: Why taking out the second argument? Should be called "pedestal_events_id" or something like that (can be interleaved or not; if not, by the way, there is no guarantee that the stars are in the right position). And then that should be used inside get_unusable_pixels where you now use INTERLEAVED_CALIBRATION_ID

File: lstchain/calib/camera/pixel_threshold_estimation.py (hunk 2 above)
Comment (pawel21) [reply lstchain/calib/camera/pixel_threshold_estimation.py]: Done

Comment (pawel21) [reply lstchain/calib/camera/pixel_threshold_estimation.py]: Fixed

File: lstchain/calib/camera/pixel_threshold_estimation.py (hunk 3 above)
Comment (pawel21) [reply lstchain/calib/camera/pixel_threshold_estimation.py]: Done

**Review by moralejo — APPROVED** [review]
Looks fine to me.

**Review by maxnoe — COMMENTED** [review]
The ids don't actually refer to events but to a set of calibration coefficients, right?

So I'd suggest using `pedestal_id` or `pedestal_calib_id` instead of `pedestal_events_id`.

**Comment by moralejo** [comment]
> The ids don't actually refer to events but to a set of calibration coefficients, right?
> 
> So I'd suggest using `pedestal_id` or `pedestal_calib_id` instead of `pedestal_events_id`.

Good point, indeed. We should keep events for camera events.

**Comment by pawel21** [comment]
> The ids don't actually refer to events but to a set of calibration coefficients, right?
> 
> So I'd suggest using `pedestal_id` or `pedestal_calib_id` instead of `pedestal_events_id`.

Done

**Review by moralejo — APPROVED** [review]

**Pull request merged**